### PR TITLE
RSBTB-2262 Add image transform

### PR DESCRIPTION
**Pull request by shandak**

### CODE REVIEW

File: libraries/redcore/api/hal/transform/image.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	public static function toInternal($definition)
+	{
+		// This is already converted to image type so we are returning it as is
+		if (!is_array($definition))
+		{
+			return $definition;
+		}
+
+		$fullPath = $definition['name'];
+		$fileName = $definition['name'];
```
Comment (Kixo) [inline libraries/redcore/api/hal/transform/image.php]: path and name comes from the same "name"?

Comment (shandak) [inline libraries/redcore/api/hal/transform/image.php]: In `$definition['name']` full web url, and `RedshopbHelperThumbnail::savingImage` can get name file and path from url

File: libraries/redcore/api/hal/transform/image.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	 */
+	public static function toInternal($definition)
+	{
+		// This is already converted to image type so we are returning it as is
+		if (!is_array($definition))
+		{
+			return $definition;
+		}
+
+		$fullPath = $definition['name'];
```
Comment (Kixo) [inline libraries/redcore/api/hal/transform/image.php]: why not call the keys in the array as they should be called: fullPath and fileName and id, and not name and section?

Comment (shandak) [inline libraries/redcore/api/hal/transform/image.php]: I will change name to fullPath.
`section` is required for function.
Use `id` most desirable

**Pull request closed without merging**